pkg/virtctl/create/instancetype/instancetype.go

```diff
@@ -0,0 +1,263 @@
+		Example: c.usage(),
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return c.run(cmd)
+		},
+	}
+	cmd.Flags().StringVar(&c.name, NameFlag, c.name, "Specify the name of the Instancetype.")
+	cmd.Flags().IntVar(&c.cpu, CPUFlag, c.cpu, "Specify the count of CPUs of the Instancetype.")
+	cmd.Flags().StringVar(&c.memory, MemoryFlag, c.memory, "Specify the amount of memory of the Instancetype.")
+	cmd.Flags().StringVar(&c.ioThreadsPolicy, IOThreadsPolicyFlag, c.ioThreadsPolicy, "Specify IOThreadsPolicy to be used. Only valid values are \"auto\" and \"shared\".")
+	cmd.Flags().BoolVar(&c.cluster, ClusterFlag, false, "Specify if clusterInstancetype should be created. By default instancetype is created.")
+
+	cmd.Flags().StringArrayVar(&c.gpus, GPUFlag, c.gpus, "Specify the list of vGPUs to passthrough. Can be provided multiple times.")
+	cmd.Flags().StringArrayVar(&c.hostDevices, HostDeviceFlag, c.hostDevices, "Specify list of HostDevices to passthrough. Can be provided multiple times.")
+	return cmd
+}
+
+func withGPUs(c *instancetype, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	for _, param := range c.gpus {
+		gpu := GPU{}
+
+		err := params.MapParams(GPUFlag, param, &gpu)
+		if err != nil {
+			return err
+		}
+
+		if gpu.Name == "" {
+			return params.FlagErr(GPUFlag, "name must be specified")
+		}
+
+		if gpu.DeviceName == "" {
+			return params.FlagErr(GPUFlag, "deviceName must be specified")
+		}
+
+		instancetypeSpec.GPUs = append(instancetypeSpec.GPUs, v1.GPU{Name: gpu.Name, DeviceName: gpu.DeviceName})
+	}
+
+	return nil
+}
+
+func withHostDevices(c *instancetype, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	for _, param := range c.hostDevices {
+		hostDevice := HostDevice{}
+
+		err := params.MapParams(HostDeviceFlag, param, &hostDevice)
+		if err != nil {
+			return err
+		}
+
+		if hostDevice.Name == "" {
+			return params.FlagErr(HostDeviceFlag, "name must be specified")
+		}
+
+		if hostDevice.DeviceName == "" {
+			return params.FlagErr(HostDeviceFlag, "deviceName must be specified")
+		}
+
+		instancetypeSpec.HostDevices = append(instancetypeSpec.HostDevices, v1.HostDevice{Name: hostDevice.Name, DeviceName: hostDevice.DeviceName})
+	}
+
+	return nil
+}
+
+func withIOThreadsPolicy(c *instancetype, instancetypeSpec *instancetypev1alpha2.VirtualMachineInstancetypeSpec) error {
+	var instancetypePolicy v1.IOThreadsPolicy
+
+	switch c.ioThreadsPolicy {
+	case string(v1.IOThreadsPolicyAuto):
+		instancetypePolicy = v1.IOThreadsPolicyAuto
+	case string(v1.IOThreadsPolicyShared):
+		instancetypePolicy = v1.IOThreadsPolicyShared
+	default:
+		return params.FlagErr(IOThreadsPolicyFlag, "IOThread must be of value auto or shared")
+	}
+
+	instancetypeSpec.IOThreadsPolicy = &instancetypePolicy
+	return nil
+}
+
+func defaultCreateInstancetype() instancetype {
+	return instancetype{
+		name:   "instancetype-" + rand.String(5),
+		cpu:    1,
+		memory: "128Mi",
```

Comment:
We shouldn't default anything here, these need to be required for each instance type.

Reply:
I suggest setting ClusterFlag to `true` here.